app/streams/publishing_api/message_adapter.rb

```diff
@@ -1,21 +1,48 @@
 module PublishingAPI
   class MessageAdapter
-    def self.to_dimension_item(*args)
-      new(*args).to_dimension_item
-    end
-
     def initialize(message)
       @message = message
     end
 
-    def to_dimension_item
-      Dimensions::Item.new(
-        content_id: message.payload.fetch('content_id'),
-        base_path: message.payload.fetch('base_path'),
+    def existing_dimension_items
```

Comment:
This method doesn't feel like part of the adapter. I would move it to the Handler, although I might be missing something.